capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,203 @@
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, restaurantKey),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year),
+                        new FilterPredicate("week", Query.FilterOperator.EQUAL, week)
+                ));
+
+        Query query = new Query("PageViews").setFilter(compositeFilter);
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        Entity results = preparedQuery.asSingleEntity();
+
+        // If there are no entities of that restaurant, return a page view with zeroes in all fields
+        if (results == null) {
+            return new WeeklyPageView(0, 0, 0);
+        }
+
+        WeeklyPageView weeklyPageView = new WeeklyPageView(
+                week,
+                year,
+                Integer.parseInt(results.getProperty("count").toString()));
+
+        return weeklyPageView;
+    }
+
+    /**
+     * Gets the weekly pageViews for the entire year of a specific restaurant.
+     * @param year we are looking for
+     * @param restaurantKey of restaurant we want the data.
+     * @return List<WeeklyPageViews> sorted by week
+     */
+    public List<WeeklyPageView> getYearRestaurantPageViews(int year, String restaurantKey) {
+
+        CompositeFilter compositeFilter = new CompositeFilter(Query.CompositeFilterOperator.AND,
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, restaurantKey),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year)
+                ));
+        Query query = new Query("PageViews").setFilter(compositeFilter).addSort("week", Query.SortDirection.ASCENDING);
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withDefaults());
+
+        List<WeeklyPageView> pageViewsList = new ArrayList<>();
+
+        WeeklyPageView pageView;
+        for (Entity entity : results) {
+            pageViewsList.add(new WeeklyPageView(Integer.parseInt(entity.getProperty("week").toString()),
+                    year,
+                    Integer.parseInt(entity.getProperty("count").toString())));
+        }
+
+        return pageViewsList;
+    }
+
+    /**
+     * Gets all pageViews for the entire system.
+     * @return returns List<RestaurantPageViews> with the PageView list inside each instance
+     *         of RestaurantPageViews sorted by week and year in ascending order.
+     */
+    public List<RestaurantPageViews> getAllPageViews() {
+        // Query with double sort so the first element of every restaurant should be
+        // the earliest week and year.
+        Query query = new Query("PageViews")
+                .addSort("week", Query.SortDirection.ASCENDING)
+                .addSort("year", Query.SortDirection.ASCENDING);
```

Comment:
Does the order of this sort matter? I'm wondering if we want to first sort by year, then by week.

Reply:
Yup you're right! Changed.